### PR TITLE
Removed lots of legacy draw calls

### PR DESCRIPTION
**Pull request by davidpiuva**
Removed legacy `draw` methods from the GPU. Point and line drawing is still holding back the removal of focal length and transform which should be moved to `Pen`.

### CODE REVIEW

File: source/draw/Canvas.ooc [hunk 1]
```diff
@@ -91,7 +91,7 @@ Canvas: abstract class {
 		}
 		message free(Owner Receiver)
 	}
-	draw: abstract func ~ImageSourceDestination (image: Image, source, destination: IntBox2D)
+	draw: virtual func ~ImageSourceDestination (image: Image, source, destination: IntBox2D) { Debug error("draw~ImageSourceDestination unimplemented for class " + this class name + "!") }
```
Comment (davidpiuva) [inline source/draw/Canvas.ooc]: Still exists for the CPU.

File: source/draw/Canvas.ooc [hunk 2]
```diff
@@ -91,8 +91,7 @@ Canvas: abstract class {
...
 	draw: func ~ImageDestination (image: Image, destination: IntBox2D) { this draw(image, IntBox2D new(image size), destination) }
 	draw: func ~Image (image: Image) { this draw(image, IntBox2D new(image size)) }
-	draw: func ~ImageTargetSize (image: Image, targetSize: IntVector2D) { this draw(image, IntBox2D new(targetSize)) }
```
Comment (davidpiuva) [inline source/draw/Canvas.ooc]: No longer used.

File: source/draw/gpu/GpuCanvas.ooc [hunk 3]
```diff
@@ -61,30 +61,6 @@ GpuCanvas: abstract class extends Canvas {
...
-		if (temporary instanceOf(GpuYuv420Semiplanar)) {
-			yuv := temporary as GpuYuv420Semiplanar
-			map add("texture0", yuv y)
-			map add("texture1", yuv uv)
-		} else
-			map add("texture0", temporary)
-		this draw(temporary, source, destination, map)
-		if (image != temporary)
-			temporary free()
-	}
```
Comment (davidpiuva) [inline source/draw/gpu/GpuCanvas.ooc]: This huge mess can be deleted now.

**Comment by davidpiuva** [comment]
@sebastianbaginski Peer review?

File: source/draw/gpu/GpuCanvas.ooc (hunk 3 above)
Comment (marcusnaslund) [inline source/draw/gpu/GpuCanvas.ooc]: :+1:

**Pull request merged**